src/blueprints/helpers.py

```
@@ -100,17 +128,20 @@ def get_kosync_id_for_ebook(ebook_filename, booklore_id=None, original_filename=
+    if booklore_id:
+        for bl_client in get_booklore_clients():
+            if not bl_client.is_configured():
+                continue
+            try:
+                content = bl_client.download_book(booklore_id)
+                if content:
+                    kosync_id = container.ebook_parser().get_kosync_id_from_bytes(ebook_filename, content)
+                    if kosync_id:
+                        logger.debug(f"Computed KOSync ID from Booklore download: '{kosync_id}'")
+                        return kosync_id
+            except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)